docs/data/tree-view/overview/CustomIcons.js

```diff
@@ -0,0 +1,89 @@
+import Box from '@mui/material/Box';
+import SvgIcon from '@mui/material/SvgIcon';
+
+import { styled } from '@mui/material/styles';
+import { TreeView } from '@mui/x-tree-view/TreeView';
+import { TreeItem, treeItemClasses } from '@mui/x-tree-view/TreeItem';
+
+const CustomTreeItem = styled(TreeItem)(({ theme }) => ({
+  [`& .${treeItemClasses.content}`]: {
+    padding: theme.spacing(0.5, 1),
+    margin: theme.spacing(0.2, 0),
+    [`& .${treeItemClasses.label}`]: {
+      fontFamily: 'monospace',
+      fontSize: '0.8rem',
+    },
+  },
+  [`& .${treeItemClasses.iconContainer}`]: {
+    '& .close': {
+      opacity: 0.3,
+    },
+  },
+  [`& .${treeItemClasses.group}`]: {
+    marginLeft: 15,
+    paddingLeft: 18,
+  },
+}));
+
+function MinusSquare(props) {
+  return (
+    <SvgIcon fontSize="inherit" style={{ width: 14, height: 14 }} {...props}>
+      {/* tslint:disable-next-line: max-line-length */}
+      <path d="M22.047 22.074v0 0-20.147 0h-20.12v0 20.147 0h20.12zM22.047 24h-20.12q-.803 0-1.365-.562t-.562-1.365v-20.147q0-.776.562-1.351t1.365-.575h20.147q.776 0 1.351.575t.575 1.351v20.147q0 .803-.575 1.365t-1.378.562v0zM17.873 11.023h-11.826q-.375 0-.669.281t-.294.682v0q0 .401.294 .682t.669.281h11.826q.375 0 .669-.281t.294-.682v0q0-.401-.294-.682t-.669-.281z" />
+    </SvgIcon>
+  );
+}
+
+function PlusSquare(props) {
+  return (
+    <SvgIcon fontSize="inherit" style={{ width: 14, height: 14 }} {...props}>
+      {/* tslint:disable-next-line: max-line-length */}
+      <path d="M22.047 22.074v0 0-20.147 0h-20.12v0 20.147 0h20.12zM22.047 24h-20.12q-.803 0-1.365-.562t-.562-1.365v-20.147q0-.776.562-1.351t1.365-.575h20.147q.776 0 1.351.575t.575 1.351v20.147q0 .803-.575 1.365t-1.378.562v0zM17.873 12.977h-4.923v4.896q0 .401-.281.682t-.682.281v0q-.375 0-.669-.281t-.294-.682v-4.896h-4.923q-.401 0-.682-.294t-.281-.669v0q0-.401.281-.682t.682-.281h4.923v-4.896q0-.401.294-.682t.669-.281v0q.401 0 .682.281t.281.682v4.896h4.923q.401 0 .682.281t.281.682v0q0 .375-.281.669t-.682.294z" />
+    </SvgIcon>
+  );
+}
+
+function CloseSquare(props) {
+  return (
+    <SvgIcon
+      className="close"
+      fontSize="inherit"
+      style={{ width: 14, height: 14 }}
+      {...props}
+    >
+      {/* tslint:disable-next-line: max-line-length */}
+      <path d="M17.485 17.512q-.281.281-.682.281t-.696-.268l-4.12-4.147-4.12 4.147q-.294.268-.696.268t-.682-.281-.281-.682.294-.669l4.12-4.147-4.12-4.147q-.294-.268-.294-.669t.281-.682.682-.281.696 .268l4.12 4.147 4.12-4.147q.294-.268.696-.268t.682.281 .281.669-.294.682l-4.12 4.147 4.12 4.147q.294.268 .294.669t-.281.682zM22.047 22.074v0 0-20.147 0h-20.12v0 20.147 0h20.12zM22.047 24h-20.12q-.803 0-1.365-.562t-.562-1.365v-20.147q0-.776.562-1.351t1.365-.575h20.147q.776 0 1.351.575t.575 1.351v20.147q0 .803-.575 1.365t-1.378.562v0z" />
+    </SvgIcon>
+  );
+}
+
+export default function CustomIcons() {
+  return (
+    <Box sx={{ minHeight: 270, flexGrow: 1, maxWidth: 300 }}>
+      <TreeView
```

Comment:
I'm wondering if we really need this Box component there in all of the demos — can't we just pass these properties to the Tree View's `sx` prop instead? Minus one component import 😃

Reply:
I thought I answer this comment, sorry if I already answer somewhere else.

For me the main benefit of the `Box` is that the preview removes it, so we end up with a smaller preview than if we have the `sx` on the `TreeView itself.

Also take into account that `TreeVIew` previews are often very close from the maximum length accepted for preview, so 1 line can make all the difference between having and not having a preview.

So, for demos where the preview contains the important piece of information, I think the `Box` is important.
For demos where the preview is pointless, clearly the `Box` is useless.

Is the consistency more important than the optimal solution in both cases? I don't know tbh.

Reply:
That makes sense! I was more optimizing for the use case where I copy the entire demo code snippet to my codebase, which, in that case, I'd have to remove either the `sx` prop on the Tree View or the Box wrapper (plus import) because these styles are specific for it to work within the docs demo container. 

It's fine for me to use this guiding rule, though: 
> if the preview contains an important piece of information, I think the Box is important. For demos where the preview is pointless, the Box is useless.

Reply:
Removed the `Box` based on the rule of thumb from above :)